### PR TITLE
feat: Introduce settlement report in language in .NET

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## References

* #199

### CODE REVIEW

File: source/dotnet/Services/Application/SettlementReport/SettlementReportApplicationService.cs [hunk 1]
```diff
@@ -13,36 +13,36 @@
...
+    public SettlementReportApplicationService(IBatchRepository batchRepository, ISettlementReportRepository settlementReportRepository, IUnitOfWork unitOfWork)
     {
         _batchRepository = batchRepository;
-        _processOutputRepository = processOutputRepository;
+        _settlementReportRepository = settlementReportRepository;
         _unitOfWork = unitOfWork;
     }
 
-    public async Task ZipBasisDataAsync(BatchCompletedEventDto batchCompletedEvent)
+    public async Task CreateSettlementReportAsync(BatchCompletedEventDto batchCompletedEvent)
```
Comment (lasrinnil) [inline source/dotnet/Services/Application/SettlementReport/SettlementReportApplicationService.cs:34]: Do we only create a single Settlement rapport?

File: source/dotnet/Services/Domain/SettlementReportAggregate/ISettlementReportRepository.cs [hunk 2]
```diff
@@ -14,15 +14,15 @@
...
 
-public interface IProcessOutputRepository
+public interface ISettlementReportRepository
 {
     /// <summary>
     /// Create zip archives for each process in the batch.
     /// The archive contains the basis data files and the result file.
     /// </summary>
-    Task CreateBasisDataZipAsync(Batch completedBatch);
+    Task CreateSettlementReportAsync(Batch completedBatch);
```
Comment (lasrinnil) [inline source/dotnet/Services/Domain/SettlementReportAggregate/ISettlementReportRepository.cs:25]: Consider moving this to a domain service in a future PR :)

**Review by lasrinnil — DISMISSED** [review]

File: source/dotnet/Services/Domain/SettlementReportAggregate/ISettlementReportRepository.cs (hunk 2 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/Domain/SettlementReportAggregate/ISettlementReportRepository.cs:25]: Yes, perhaps. I'll, however, revisit it when we start creating multiple settlement reports.

File: source/dotnet/Services/Application/SettlementReport/SettlementReportApplicationService.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/Application/SettlementReport/SettlementReportApplicationService.cs:34]: Yes, but we _should_ create multiple. I'll rename to plural.

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**